src/chain/stateTransition/block/attestations.ts

```diff
@@ -23,61 +21,84 @@ import {
+// if data.target_epoch == get_current_epoch(state):
+// ffg_data = (state.current_justified_epoch, state.current_justified_root, get_current_epoch(state))
+// parent_crosslink = state.current_crosslinks[data.crosslink.shard]
+// state.current_epoch_attestations.append(pending_attestation)
+// else:
+// ffg_data = (state.previous_justified_epoch, state.previous_justified_root, get_previous_epoch(state))
+// parent_crosslink = state.previous_crosslinks[data.crosslink.shard]
+// state.previous_epoch_attestations.append(pending_attestation)
+//
+// # Check FFG data, crosslink data, and signature
+// assert ffg_data == (data.source_epoch, data.source_root, data.target_epoch)
+// assert data.crosslink.start_epoch == parent_crosslink.end_epoch
+// assert data.crosslink.end_epoch == min(data.target_epoch, parent_crosslink.end_epoch + MAX_EPOCHS_PER_CROSSLINK)
+// assert data.crosslink.parent_root == hash_tree_root(parent_crosslink)
+// assert data.crosslink.data_root == ZERO_HASH  # [to be removed in phase 1]
+// validate_indexed_attestation(state, convert_to_indexed(state, attestation))
+
 
-export function processAttestation(state: BeaconState, attestation: Attestation): BeaconState {
+export default function processAttestation(state: BeaconState, attestation: Attestation): void {
   const currentEpoch = getCurrentEpoch(state);
   const previousEpoch = getPreviousEpoch(state);
   const data = attestation.data;
   const attestationSlot = getAttestationDataSlot(state, data);
+  let ffgData, parentCrosslink;
   assert(
     attestationSlot + MIN_ATTESTATION_INCLUSION_DELAY <= state.slot &&
     state.slot <= attestationSlot + SLOTS_PER_EPOCH
   );
 
-  // Check target epoch, source epoch, and source root
-  assert((
-    currentEpoch === data.targetEpoch &&
-    state.currentJustifiedEpoch === data.sourceEpoch &&
-    state.currentJustifiedRoot.equals(data.sourceRoot) &&
-    hashTreeRoot(state.currentCrosslinks[data.shard], Crosslink).equals(data.previousCrosslinkRoot)
-  ) || (
-    previousEpoch === data.targetEpoch &&
-    state.previousJustifiedEpoch === data.sourceEpoch &&
-    state.previousJustifiedRoot.equals(data.sourceRoot) &&
-    hashTreeRoot(state.previousCrosslinks[data.shard], Crosslink).equals(data.previousCrosslinkRoot)
-  ));
-
-  // Check crosslink data
-  assert(data.crosslinkDataRoot.equals(ZERO_HASH)); // TO BE REMOVED IN PHASE 1
-
-  // Check signature and bitfields
-  assert(verifyIndexedAttestation(state, convertToIndexed(state, attestation)));
+  // assert([previousEpoch, currentEpoch].includes(data.targetEpoch));
 
   // Cache pending attestation
   const pendingAttestation: PendingAttestation = {
-    data,
+    data: data,
     aggregationBitfield: attestation.aggregationBitfield,
     inclusionDelay: state.slot - attestationSlot,
     proposerIndex: getBeaconProposerIndex(state),
   };
 
   if (data.targetEpoch === currentEpoch) {
+    ffgData = [state.currentJustifiedEpoch, state.currentJustifiedRoot, currentEpoch];
```

Comment:
If i console log ffgData with right side value it's always same. But the comparing always gives false. what's the right way of comparing then?